src/proxyvisor.coffee

```diff
@@ -0,0 +1,211 @@
+			resinApi.post
+				resource: 'device'
+				body: device
+				customOptions:
+					apikey: apiKey
+			.then (dev) ->
+				deviceForDB = {
+					uuid: uuid
+					appId: device.application
+					device_type: device.device_type
+					deviceId: dev.id
+					name: dev.name
+					status: device.status
+					logs_channel: device.logs_channel
+				}
+				knex('dependentDevice').insert(deviceForDB)
+				.then ->
+					res.status(202).send(dev)
+	)
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.get '/v1/devices/:uuid', (req, res) ->
+	uuid = req.params.uuid
+	knex('dependentDevice').select().where({ uuid })
+	.then ([ device ]) ->
+		return res.status(404).send('Device not found') if !device?
+		res.json(device)
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.put '/v1/devices/:uuid/logs', (req, res) ->
+	uuid = req.params.uuid
+	m = {
+		message: req.body.message
+		timestamp: req.body.timestamp or Date.now()
+	}
+	m.isSystem = req.body.isSystem if req.body.isSystem?
+
+	knex('dependentDevice').select().where({ uuid })
+	.then ([ device ]) ->
+		return res.status(404).send('Device not found') if !device?
+		pubnub.publish({ channel: "device-#{device.logs_channel}-logs", message: m })
+	.then ->
+		res.status(202).send('OK')
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.put '/v1/devices/:uuid', (req, res) ->
+	uuid = req.params.uuid
+	status = req.body.status
+	is_online = req.body.is_online
+	utils.getConfig('apiKey')
+	.then (apiKey) ->
+		throw new Error('apikey not found') if !apiKey?
+		knex('dependentDevice').select().where({ uuid })
+		.then ([ device ]) ->
+			return res.status(404).send('Device not found') if !device?
+			resinApi.patch
+				resource: 'device'
+				id: device.deviceId
+				body: { status, is_online }
+				customOptions:
+					apikey: apiKey
+			.then ->
+				device.status = status
+				device.is_online = is_online
+				knex('dependentDevice').update(device).where({ uuid })
+				.then ->
+					res.json(device)
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+tarPath = (app) ->
+	return '/tmp/' + app.commit + '.tar'
+
+router.get '/v1/assets/:commit', (req, res) ->
+	knex('dependentApp').select().where({ commit: req.params.commit })
+	.then ([ app ]) ->
+		return res.status(404).send('Not found') if !app
+		dest = tarPath(app)
+		getAssetsPath(app.imageId)
+		.then (path) ->
+			getTarArchive(path, dest)
+		.then (archive) ->
+			new Promise (resolve, reject) ->
+				archive.on 'finish', ->
+					res.sendFile(dest)
+					resolve()
+				archive.on 'error', (err) ->
+					reject(err)
+	.catch (err) ->
+		console.error(err)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+getTarArchive = (path, destination) ->
+	fs.lstatAsync(path)
+	.then ->
+		tarArchive = fs.createWriteStream(destination)
+		tar.pack(path).pipe tarArchive
```

Comment:
I expect this needs some `.on 'error', ->` handlers in order to not crash nodejs on stream errors